### PR TITLE
PRAC-41 Add/remove user to/from activity

### PR DESCRIPTION
**Pull request by vitaliipp**
Add user to activity route and controller has been created.
Endpoint: http://localhost:8000/api/v1/activities/addMe/activityId

Remove user from activity route and controller has been created.
Endpoint: http://localhost:8000/api/v1/activities/removeMe/activityId

Routes have been tested with Postman.

### CODE REVIEW

File: src/controllers/activities.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+    res.status(StatusCodes.OK).json({ activities, count: activities.length });
+  } catch (error) {
+    throw new BadRequestError('Error with getAllActivites');
+  }
+};
 
 const getActivity = async (req, res) => {
   const {
-    user: { userId },
+    //    user: { userId },
```
Comment (benochi) [inline src/controllers/activities.js:16]: Are we allowing anyone to get activities by ID or how are we handling this?

File: src/controllers/activities.js [hunk 2]
```diff
@@ -99,13 +103,51 @@ const deleteActivity = async (req, res) => {
   if (!activity) {
     throw new NotFoundError(`No activity with id ${activityId}`);
   }
-  res.status(StatusCodes.OK).json({ msg: "Activity was deleted" });
+  res.status(StatusCodes.OK).json({ msg: 'Activity was deleted' });
+};
+
+const addUserToActivity = async (req, res) => {
+  const { id: activityId } = req.params;
+  const userId = req.user.userId;
```
Comment (benochi) [inline src/controllers/activities.js:111]: You should not trust the user to send their ID from the client side, is a middleware applying this to the user or is this from the request itself?

**Review by benochi — CHANGES_REQUESTED** [review]
Can you just address my comments and let me know, If those are the ways we want these to function it looks fine, or if these are things to be addressed in the future?

File: src/controllers/activities.js (hunk 2 above)
Comment (benochi) [reply src/controllers/activities.js:111]: To further explain, are we letting any user add any other user to activities, or do we want them to only add themselves?  If so, you should get the userID from their JWT, or else anyone can make a req.user.userID with other usersIds' to this endpoint.  Not a huge deal for the mvp, but it's a security issue in the real world. If you need more info, let me know.

Comment (vitaliipp) [reply src/controllers/activities.js:111]: You are absolutely right. I've added authentication middleware into the activity router, enabling us to retrieve the userId from this middleware.

router.route('/addMe/:id').patch(authenticateUser, addUserToActivity);
router.route('/removeMe/:id').patch(authenticateUser, removeUserFromActivity);

**Comment by vitaliipp** [comment]
You are absolutely right. I've added authentication middleware into the activity router, enabling us to retrieve the userId from this middleware.

router.route('/addMe/:id').patch(authenticateUser, addUserToActivity);
router.route('/removeMe/:id').patch(authenticateUser, removeUserFromActivity);

**Review by benochi — APPROVED** [review]

**Review by Ljanjic — APPROVED** [review]

**Pull request merged**